### PR TITLE
Try an experiment

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please benchmark

**Comment by airspeedswift** [comment]
Sooo the fast path is slower?

**Comment by Catfish-Man** [comment]
@swift-ci please benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Comment by Catfish-Man** [comment]
Reminder for myself next time: you're waiting on Michael to figure out the compiler bug it's hitting

/Users/david/swift-external/build/Ninja-ReleaseAssert+stdlib-Release/swift-macosx-x86_64/stdlib/public/core/8/Tuple.swift
---
4.	While evaluating request IRGenRequest(IR Generation for module Swift)
5.	While evaluating request ExecuteSILPipelineRequest(Run pipelines { IRGen Preparation } on SIL for Swift)
6.	While running pass #19018 SILFunctionTransform "StackSizeLimitTransform" on SILFunction "@$ss6SIMD32V16debugDescriptionSSvg".
 for getter for debugDescription (at /Users/david/swift-external/build/Ninja-ReleaseAssert+stdlib-Release/swift-macosx-x86_64/stdlib/public/core/8/SIMDVectorTypes.swift:1022:14)
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x0000000107edf9d7 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 39
1  swift-frontend           0x0000000107ede8e8 llvm::sys::RunSignalHandlers() + 248
2  swift-frontend           0x0000000107ee0236 SignalHandler(int) + 278
3  libsystem_platform.dylib 0x00007ff803dbd06d _sigtramp + 29
4  libsystem_platform.dylib 0x0000000000000001 _sigtramp + 18446603370516197297
5  libsystem_c.dylib        0x00007ff803cf9d10 abort + 123
6  libsystem_c.dylib        0x00007ff803cf90be err + 0
7  swift-frontend           0x0000000107f3aba3 swift::SILBuilder::createAllocBox(swift::SILLocation, swift::CanTypeWrapper<swift::SILBoxType>, llvm::Optional<swift::SILDebugVariable>, bool) (.cold.1) + 35
8  swift-frontend           0x00000001034da3f2 swift::SILBuilder::createAllocBox(swift::SILLocation, swift::CanTypeWrapper<swift::SILBoxType>, llvm::Optional<swift::SILDebugVariable>, bool) + 386
9  swift-frontend           0x000000010381daa3 swift::SILBuilder::createAllocBox(swift::SILLocation, swift::SILType, llvm::Optional<swift::SILDebugVariable>, bool) + 227
10 swift-frontend           0x000000010381d777 convertStackBoxToHeapBox(swift::AllocStackInst*, llvm::ilist_iterator<llvm::ilist_detail::node_options<swift::SILInstruction, true, false, void>, false, false>) + 183
11 swift-frontend           0x000000010381ce38 (anonymous namespace)::StackSizeLimitTransform::run() + 248
12 swift-frontend           0x0000000103c3d4ae swift::SILPassManager::runPassOnFunction(unsigned int, swift::SILFunction*) + 1822
13 swift-frontend           0x0000000103c3e3da swift::SILPassManager::runFunctionPasses(unsigned int, unsigned int) + 1386
14 swift-frontend           0x0000000103c3b218 swift::SILPassManager::executePassPipelinePlan(swift::SILPassPipelinePlan const&) + 72
15 swift-frontend           0x0000000103c3b1b3 swift::ExecuteSILPipelineRequest::evaluate(swift::Evaluator&, swift::SILPipelineExecutionDescriptor) const + 51
16 swift-frontend           0x0000000103c5bfed swift::SimpleRequest<swift::ExecuteSILPipelineRequest, std::__1::tuple<> (swift::SILPipelineExecutionDescriptor), (swift::RequestFlags)1>::evaluateRequest(swift::ExecuteSILPipelineRequest const&, swift::Evaluator&) + 29
17 swift-frontend           0x0000000103c47481 llvm::Expected<swift::ExecuteSILPipelineRequest::OutputType> swift::Evaluator::getResultUncached<swift::ExecuteSILPipelineRequest>(swift::ExecuteSILPipelineRequest const&) + 241
18 swift-frontend           0x0000000103c3b444 swift::executePassPipelinePlan(swift::SILModule*, swift::SILPassPipelinePlan const&, bool, swift::irgen::IRGenModule*) + 68
19 swift-frontend           0x0000000103789191 swift::IRGenRequest::evaluate(swift::Evaluator&, swift::IRGenDescriptor) const + 2209
20 swift-frontend           0x00000001037c2efc swift::GeneratedModule swift::SimpleRequest<swift::IRGenRequest, swift::GeneratedModule (swift::IRGenDescriptor), (swift::RequestFlags)9>::callDerived<0ul>(swift::Evaluator&, std::__1::integer_sequence<unsigned long, 0ul>) const + 204
21 swift-frontend           0x00000001037c2dee swift::SimpleRequest<swift::IRGenRequest, swift::GeneratedModule (swift::IRGenDescriptor), (swift::RequestFlags)9>::evaluateRequest(swift::IRGenRequest const&, swift::Evaluator&) + 14
22 swift-frontend           0x00000001037941b5 llvm::Expected<swift::IRGenRequest::OutputType> swift::Evaluator::getResultUncached<swift::IRGenRequest>(swift::IRGenRequest const&) + 389
23 swift-frontend           0x000000010378a2d9 swift::performIRGeneration(swift::ModuleDecl*, swift::IRGenOptions const&, swift::TBDGenOptions const&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule> >, llvm::StringRef, swift::PrimarySpecificPaths const&, llvm::ArrayRef<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, llvm::GlobalVariable**) + 649
24 swift-frontend           0x0000000103333919 performCompileStepsPostSILGen(swift::CompilerInstance&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule> >, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, swift::PrimarySpecificPaths const&, int&, swift::FrontendObserver*) + 1705
25 swift-frontend           0x000000010333302d performCompileStepsPostSema(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 685
26 swift-frontend           0x000000010332757d swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 4637
27 swift-frontend           0x00000001032d1dd3 swift::mainEntry(int, char const**) + 499
28 dyld                     0x00000001146d24d5 start + 421
29 dyld                     000000000000000000 start + 18446744069071887568
30 swift-frontend           0x00000001032af000 __dso_handle + 0
[26/712][  3%][142.157s] Generating /Users/david/swift-external/b.../lib/swift/macosx/Swift.swiftmodule/arm64-apple-macos.swiftmodule

File: include/swift/AST/Builtins.def [hunk 1]
```diff
@@ -880,6 +880,11 @@ BUILTIN_MISC_OPERATION_WITH_SILGEN(BuildDefaultActorExecutorRef,
 BUILTIN_MISC_OPERATION_WITH_SILGEN(BuildMainActorExecutorRef,
                                    "buildMainActorExecutorRef", "n", Special)
 
+/// isOnStack : (Builtin.NativeObject) -> Builtin.Int1
+///
+/// Returns true if the optimizer successfully moved this object from the stack onto the heap.
```
Comment (tbkka) [inline include/swift/AST/Builtins.def:885]: Is this comment backwards?

Comment (Catfish-Man) [reply include/swift/AST/Builtins.def:885]: I think so but you'll have to ask @gottesmm since it's from his PR that I merged into this one

**Pull request closed without merging**